media-libs/embree/embree-3.13.0.ebuild

```diff
@@ -0,0 +1,107 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake flag-o-matic linux-info toolchain-funcs
+
+DESCRIPTION="Collection of high-performance ray tracing kernels"
+HOMEPAGE="https://github.com/embree/embree"
+LICENSE="Apache-2.0"
+KEYWORDS="~amd64 ~x86"
+SRC_URI="https://github.com/embree/embree/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+SLOT="3"
+X86_CPU_FLAGS=( sse2:sse2 sse4_2:sse4_2 avx:avx avx2:avx2 avx512dq:avx512dq )
+CPU_FLAGS=( ${X86_CPU_FLAGS[@]/#/cpu_flags_x86_} )
+IUSE="+compact-polys ispc raymask -ssp +tbb tutorial static-libs ${CPU_FLAGS[@]%:*}"
+BDEPEND="
+	virtual/pkgconfig
+"
+RDEPEND="
+	ispc? ( dev-lang/ispc )
+	>=media-libs/glfw-3.2.1
+	tbb? ( dev-cpp/tbb )
+	tutorial? (
+		>=media-libs/libpng-1.6.34:0=
+		>=media-libs/openimageio-1.8.7
+		virtual/jpeg:0
+	)
+	virtual/opengl
+"
+DEPEND="${RDEPEND}"
+RESTRICT="mirror"
+DOCS=( CHANGELOG.md README.md readme.pdf )
+
+pkg_setup() {
+	CONFIG_CHECK="~TRANSPARENT_HUGEPAGE"
+	WARNING_TRANSPARENT_HUGEPAGE="Not enabling Transparent Hugepages (CONFIG_TRANSPARENT_HUGEPAGE) will impact rendering performance."
+	linux-info_pkg_setup
+
+	if ! ( cat /proc/cpuinfo | grep sse2 > /dev/null ) ; then
+		die "You need a CPU with at least sse2 support"
+	fi
+}
+
+src_prepare() {
+	cmake_src_prepare
+
+	# disable RPM package building
+	sed -e 's|CPACK_RPM_PACKAGE_RELEASE 1|CPACK_RPM_PACKAGE_RELEASE 0|' \
+		-i CMakeLists.txt || die
+}
+
+src_configure() {
+	# NOTE: You can make embree accept custom CXXFLAGS by turning off
+	# EMBREE_IGNORE_CMAKE_CXX_FLAGS. However, the linking will fail if you use
+	# any "march" compile flags. This is because embree builds modules for the
+	# different supported ISAs and picks the correct one at runtime.
+	# "march" will pull in cpu instructions that shouldn't be in specific modules
+	# and it fails to link properly.
+	# https://github.com/embree/embree/issues/115
+
+	filter-flags -march=*
+
+	local mycmakeargs=(
+		-DBUILD_TESTING:BOOL=OFF
```

Comment:
Could we have testing in future? $(usex test)

Reply:
We don't have access the the tests.
It seems from the files that they host all the test files on local servers...

I've added a comment about this in my next pull request.